power_grid_model_c/power_grid_model/include/power_grid_model/math_solver/math_solver_dispatch.hpp

```diff
@@ -0,0 +1,156 @@
+              }},
+              run_state_estimation{[](void* solver, StateEstimationInput<sym> const& input, double err_tol,
+                                      Idx max_iter, CalculationInfo& calculation_info,
+                                      CalculationMethod calculation_method, YBus<sym> const& y_bus) {
+                  return reinterpret_cast<MathSolverType<sym>*>(solver)->run_state_estimation(
+                      input, err_tol, max_iter, calculation_info, calculation_method, y_bus);
+              }},
+              run_short_circuit{[](void* solver, ShortCircuitInput const& input, CalculationInfo& calculation_info,
+                                   CalculationMethod calculation_method, YBus<sym> const& y_bus) {
+                  return reinterpret_cast<MathSolverType<sym>*>(solver)->run_short_circuit(input, calculation_info,
+                                                                                           calculation_method, y_bus);
+              }},
+              clear_solver{[](void* solver) { reinterpret_cast<MathSolverType<sym>*>(solver)->clear_solver(); }},
+              parameters_changed{[](void* solver, bool changed) {
+                  reinterpret_cast<MathSolverType<sym>*>(solver)->parameters_changed(changed);
+              }} {}
+
+        std::add_pointer_t<void*(std::shared_ptr<MathModelTopology const> const&)> create;
+        std::add_pointer_t<void(void const*)> destroy;
+        std::add_pointer_t<void*(void const*)> copy;
+        std::add_pointer_t<SolverOutput<sym>(void*, PowerFlowInput<sym> const&, double, Idx, CalculationInfo&,
+                                             CalculationMethod, YBus<sym> const&)>
+            run_power_flow;
+        std::add_pointer_t<SolverOutput<sym>(void*, StateEstimationInput<sym> const&, double, Idx, CalculationInfo&,
+                                             CalculationMethod, YBus<sym> const&)>
+            run_state_estimation;
+        std::add_pointer_t<ShortCircuitSolverOutput<sym>(void*, ShortCircuitInput const&, CalculationInfo&,
+                                                         CalculationMethod, YBus<sym> const&)>
+            run_short_circuit;
+        std::add_pointer_t<void(void*)> clear_solver;
+        std::add_pointer_t<void(void*, bool)> parameters_changed;
+    };
+
+    template <template <symmetry_tag> class MathSolverType>
+    constexpr MathSolverDispatcher(math_solver_tag<MathSolverType> /* unused */)
+        : sym_config{math_solver_tag<MathSolverType>{}}, asym_config{math_solver_tag<MathSolverType>{}} {}
+
+    template <symmetry_tag sym> MathSolverDispatcherConfig<sym> const& get_dispather_config() const {
+        if constexpr (is_symmetric_v<sym>) {
+            return sym_config;
+        } else {
+            return asym_config;
+        }
+    }
+    MathSolverDispatcherConfig<symmetric_t> sym_config;
+    MathSolverDispatcherConfig<asymmetric_t> asym_config;
```

Comment:
probably better to make these `private` since you have a `get_dispatcher_config()` entrypoint anyway
```suggestion

  private:
    MathSolverDispatcherConfig<symmetric_t> sym_config;
    MathSolverDispatcherConfig<asymmetric_t> asym_config;
```